datanode/api/server.go

```diff
@@ -80,6 +81,50 @@ type NetworkHistoryService interface {
 	GetBootstrapPeers() []string
 }
 
+// MarketsService ...
+//
+//go:generate go run github.com/golang/mock/mockgen -destination mocks/markets_service_mock.go -package mocks code.vegaprotocol.io/vega/datanode/api MarketsService
+type MarketsService interface {
+	Upsert(ctx context.Context, market *entities.Market) error
+	GetByID(ctx context.Context, marketID string) (entities.Market, error)
+	GetByTxHash(ctx context.Context, txHash entities.TxHash) ([]entities.Market, error)
+	GetMarketScalingFactor(ctx context.Context, marketID string) (num.Decimal, bool)
+	GetAllPaged(ctx context.Context, marketID string, pagination entities.CursorPagination, includeSettled bool) ([]entities.Market, entities.PageInfo, error)
+	ListSuccessorMarkets(ctx context.Context, marketID string, childrenOnly bool, pagination entities.CursorPagination) ([]entities.SuccessorMarket, entities.PageInfo, error)
+}
+
+// MarketDataService ...
+//
+//go:generate go run github.com/golang/mock/mockgen -destination mocks/market_data_service_mock.go -package mocks code.vegaprotocol.io/vega/datanode/api MarketDataService
+type MarketDataService interface {
+	Add(data *entities.MarketData) error
+	Flush(ctx context.Context) error
+	Initialise(ctx context.Context) error
+	GetMarketDataByID(ctx context.Context, marketID string) (entities.MarketData, error)
+	GetMarketsData(ctx context.Context) ([]entities.MarketData, error)
+	GetHistoricMarketData(ctx context.Context, marketID string, start, end *time.Time, pagination entities.Pagination) ([]entities.MarketData, entities.PageInfo, error)
+	ObserveMarketData(ctx context.Context, retries int, marketID []string) (<-chan []*entities.MarketData, uint64)
+}
+
+// RiskFactorService ...
+//
+//go:generate go run github.com/golang/mock/mockgen -destination mocks/risk_factor_service_mock.go -package mocks code.vegaprotocol.io/vega/datanode/api RiskFactorService
+type RiskFactorService interface {
+	Upsert(ctx context.Context, factor *entities.RiskFactor) error
+	GetMarketRiskFactors(ctx context.Context, marketID string) (entities.RiskFactor, error)
+}
+
+// Asset service ...
+//
+//go:generate go run github.com/golang/mock/mockgen -destination mocks/asset_service_mock.go -package mocks code.vegaprotocol.io/vega/datanode/api AssetService
+type AssetService interface {
+	Add(ctx context.Context, a entities.Asset) error
+	GetByID(ctx context.Context, id string) (entities.Asset, error)
+	GetByTxHash(ctx context.Context, txHash entities.TxHash) ([]entities.Asset, error)
+	GetAll(ctx context.Context) ([]entities.Asset, error)
+	GetAllWithCursorPagination(ctx context.Context, pagination entities.CursorPagination) ([]entities.Asset, entities.PageInfo, error)
+}
```

Comment:
Can we only add the methods that are used in here in those mocks? Not a big deal per se, but first things I've seen here when looking at the file is that the interfaces include all the write methods, and I thoughts straight away -> there's something very wrong going on with this PR if we write from the API layer.

Reply:
Fixed now